pkg/controllers/csi/provisioner/controller.go

```diff
@@ -222,74 +246,41 @@ func (provisioner *OneAgentProvisioner) updateAgentInstallation(
-		} else if updatedDigest != "" {
-			dynakubeMetadata.LatestVersion = ""
-			dynakubeMetadata.ImageDigest = updatedDigest
 		}
+
+		tenantConfig.DownloadedCodeModuleVersion = updatedDigest
 	} else {
 		updateVersion, err := provisioner.installAgentZip(ctx, *dk, dtc, latestProcessModuleConfig)
 		if err != nil {
 			log.Info("error when updating agent from zip", "error", err.Error())
 			// reporting error but not returning it to avoid immediate requeue and subsequently calling the API every few seconds
 			return true, nil
-		} else if updateVersion != "" {
-			dynakubeMetadata.LatestVersion = updateVersion
-			dynakubeMetadata.ImageDigest = ""
+		}
+
+		if updateVersion != "" {
+			tenantConfig.DownloadedCodeModuleVersion = updateVersion
 		}
 	}
 
 	return false, nil
 }
 
-func (provisioner *OneAgentProvisioner) handleMetadata(ctx context.Context, dk *dynatracev1beta1.DynaKube) (*metadata.Dynakube, metadata.Dynakube, error) {
-	dynakubeMetadata, err := provisioner.db.GetDynakube(ctx, dk.Name)
-	if err != nil {
-		return nil, metadata.Dynakube{}, errors.WithStack(err)
-	}
-
-	// In case of a new dynakubeMetadata
-	var oldDynakubeMetadata metadata.Dynakube
-	if dynakubeMetadata != nil {
-		oldDynakubeMetadata = *dynakubeMetadata
-	}
-
+func (provisioner *OneAgentProvisioner) handleMetadata(dk *dynatracev1beta1.DynaKube) (*metadata.TenantConfig, error) {
 	tenantUUID, err := dk.TenantUUIDFromApiUrl()
 	if err != nil {
-		return nil, metadata.Dynakube{}, err
+		return nil, err
 	}
 
-	dynakubeMetadata = metadata.NewDynakube(
-		dk.Name,
-		tenantUUID,
-		oldDynakubeMetadata.LatestVersion,
-		oldDynakubeMetadata.ImageDigest,
-		dk.FeatureMaxFailedCsiMountAttempts())
-
-	return dynakubeMetadata, oldDynakubeMetadata, nil
-}
-
-func (provisioner *OneAgentProvisioner) createOrUpdateDynakubeMetadata(ctx context.Context, oldDynakube metadata.Dynakube, dynakube *metadata.Dynakube) error {
-	if oldDynakube != *dynakube {
-		log.Info("dynakube has changed",
-			"name", dynakube.Name,
-			"tenantUUID", dynakube.TenantUUID,
-			"version", dynakube.LatestVersion,
-			"max mount attempts", dynakube.MaxFailedMountAttempts)
-
-		if oldDynakube == (metadata.Dynakube{}) {
-			log.Info("adding dynakube to db", "tenantUUID", dynakube.TenantUUID, "version", dynakube.LatestVersion)
-
-			return provisioner.db.InsertDynakube(ctx, dynakube)
-		} else {
-			log.Info("updating dynakube in db",
-				"old version", oldDynakube.LatestVersion, "new version", dynakube.LatestVersion,
-				"old tenantUUID", oldDynakube.TenantUUID, "new tenantUUID", dynakube.TenantUUID)
-
-			return provisioner.db.UpdateDynakube(ctx, dynakube)
-		}
+	newTenantConfig := &metadata.TenantConfig{
+		UID:                         string(dk.UID),
+		Name:                        dk.Name,
+		TenantUUID:                  tenantUUID,
+		DownloadedCodeModuleVersion: dk.CodeModulesVersion(),
+		MaxFailedMountAttempts:      int64(dk.FeatureMaxFailedCsiMountAttempts()),
+		// WIP other fields???
```

Comment:
anything to add?

Reply:
Thank you, I missed this comment
Removed!